simulation/halsim_ws_core/src/main/native/include/HALSimBaseWebSocketConnection.h

```diff
@@ -0,0 +1,25 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2020 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#pragma once
+
+#include <memory>
+
+#include <wpi/json.h>
+
+namespace wpilibws {
+
+class HALSimBaseWebSocketConnection
+    : public std::enable_shared_from_this<HALSimBaseWebSocketConnection> {
```

Comment:
This is definitely the issue.  This needs to be eliminated or there's a diamond conflict on the server side (there ends up being two enable_shared_from_this base classes).

Reply:
The easy fix for the enable_shared_from_this issue is simply to remove it from HALSimBaseWebSocketConnection, add it to HALSimWSClientConnection, and remove the qualifier from the calls to shared_from_this() in HALSimHttpConnection.cpp.  With this change, the server works fine for me.

Reply:
Thanks @PeterJohnson ! I was going to make the changes this morning but you beat me to it :)